pkg/controllers/member/serviceexport/controller.go

```diff
@@ -193,6 +195,34 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, r.MemberClient.Status().Update(ctx, &svcExport)
 	}
 
+	if exportWeight == 0 {
+		// The weight is 0, unexport the service.
+		klog.V(2).InfoS("Service has weight 0; unexport the service", "service", svcRef)
+		r.Recorder.Eventf(&svcExport, corev1.EventTypeNormal, "Service", "Service %s weight is set to 0", svc.Name)
+
+		if controllerutil.ContainsFinalizer(&svcExport, svcExportCleanupFinalizer) {
+			if _, err = r.unexportService(ctx, &svcExport); err != nil {
+				klog.ErrorS(err, "Failed to unexport the service", "service", svcRef)
+				return ctrl.Result{}, err
+			}
+		}
+		validCond := meta.FindStatusCondition(svcExport.Status.Conditions, string(fleetnetv1alpha1.ServiceExportValid))
+		expectedValidCond := metav1.Condition{
+			Type:               string(fleetnetv1alpha1.ServiceExportValid),
+			Status:             metav1.ConditionTrue,
+			Reason:             svcExportWeightZeroReason,
```

Comment:
this is a kinda confusing condition to me. So the export succeeded but its not exported...

Reply:
updated the message, the current implementation will remove the internalServiceExport in the hub. If all the clusters are set weight as 0, the trafficManagerBackend accepted condition will be false because of not found serviceImport.